src/offcanvas/offcanvas-panel.ts

```diff
@@ -0,0 +1,154 @@
+import {DOCUMENT} from '@angular/common';
+import {
+  Component,
+  ElementRef,
+  EventEmitter,
+  Inject,
+  Input,
+  NgZone,
+  OnDestroy,
+  OnInit,
+  Output,
+  ViewEncapsulation
+} from '@angular/core';
+
+import {fromEvent, Observable, Subject} from 'rxjs';
+import {filter, take, takeUntil} from 'rxjs/operators';
+
+import {getFocusableBoundaryElements} from '../util/focus-trap';
+import {Key} from '../util/key';
+import {OffcanvasDismissReasons} from './offcanvas-dismiss-reasons';
+import {ngbRunTransition, NgbTransitionOptions} from '../util/transition/ngbTransition';
+import {reflow} from '../util/util';
+
+@Component({
+  selector: 'ngb-offcanvas-panel',
+  template: '<ng-content></ng-content>',
+  encapsulation: ViewEncapsulation.None,
+  styleUrls: [],
+  host: {
+    '[class]': '"offcanvas offcanvas-" + position  + (panelClass ? " " + panelClass : "")',
+    'role': 'dialog',
+    'tabindex': '-1',
+    '[attr.aria-modal]': 'true',
+    '[attr.aria-labelledby]': 'ariaLabelledBy',
+  }
+})
+export class NgbOffcanvasPanel implements OnInit,
+    OnDestroy {
+  private _closed$ = new Subject<void>();
+  private _elWithFocus: Element | null = null;  // element that is focused prior to offcanvas opening
+
+  @Input() animation: boolean;
+  @Input() ariaLabelledBy?: string;
+  @Input() keyboard = true;
+  @Input() panelClass: string;
+  @Input() position: 'start' | 'end' | 'top' | 'bottom' = 'start';
```

Comment:
```suggestion
  @Input() position: 'start' | 'end' | 'top' | 'bottom';
```
there is no default placement for offcanvas in native implementation. Is the different behavior in NG bootstrap desired?
https://getbootstrap.com/docs/5.1/components/offcanvas/#placement

Reply:
I think it's a good choice. It's a reasonable default, that can very easily be changed, either by passing an explicit option to open(), or by setting the default in the NgbOffcanvasConfig. 
And it makes the usage consistent with modals, where there is no need to pass any option to `open()`.
The alternative would be to allow no option to be passed (because NgbOffcanvasConfig might have set a default), bit to fail at runtime if the option isn't specified in the options nor in the config. I find it more developer-friendly to just use a reasonable default value.

Reply:
thx!